install-validator/internal/dir/upgrade.go

```diff
@@ -0,0 +1,43 @@
+package dir
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/vmware-tanzu/graph-framework-for-microservices/install-validator/internal/kubernetes"
+)
+
+func ApplyDir(directory string, force bool, c kubernetes.ClientInt, cFunc compareFunc) error {
+	// check for incompatible models and not installed. Return  if any and force != true
+	incNames, _, text, err := CheckDir(directory, c, cFunc)
```

Comment:
The nonInstalled crd's returned by the CheckDir() is being ignored.

This is bug.
If CR's exist for nonInstalled CRD's, then the installation should fail.
If no CR's exits for notInstalled CRD"s, then these CRD's should be removed from the system, before upserting crd's from the new datamodel.

cc @PavitraS

Reply:
NonInstalled return should actually be deleted.
This func just returns incompatible crds and then (if force ==true) we check if we have any data.
Then we apply all the crds - doesn't need to delete anything, because the apply client.ApplyCrd func handles that for us -create or update crd, according to if it exist or not.

Reply:
The usecase is done and tested